packages/common/core-utils/src/oob.ts

```diff
@@ -0,0 +1,17 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Use this function to assert that an array index is not out-of-bounds.
+ * @example
+ * ```ts
+ * // We know that `numberArray` has four elements in it, so this is safe.
+ * const n: number = numberArray[3] ?? oob();
+ * ```
+ * @internal
+ */
+export function oob(): never {
```

Comment:
We shouldn't use abbreviations like this in API names. We should call this `outOfBounds` or something along those lines.

Reply:
Our guidance on this could probably use some clarifying, but details can be found here:
https://github.com/microsoft/FluidFramework/wiki/Coding-Guidelines#-do-use-concise-names-for-frequently-used-apis